File: Card.java
public class Card {

    private final String question;
    private final String answer;
    private Boolean lastAnswerCorrect;

    public Card(String question, String answer) {
        this.question = question;
        this.answer = answer;
        this.lastAnswerCorrect = null;
    }

    public String getQuestion() {
        return question;
    }

    public String getAnswer() {
        return answer;
    }

    public boolean checkAnswer(String userAnswer) {
        if (userAnswer == null) {
            lastAnswerCorrect = false;
            return false;
        }

        String normalizedUser = userAnswer.trim();
        boolean correct = answer.equalsIgnoreCase(normalizedUser);
        lastAnswerCorrect = correct;
        return correct;
    }

    public boolean wasIncorrectInLastRound() {
        return Boolean.FALSE.equals(lastAnswerCorrect);
    }

    @Override
    public String toString() {
        return "Card{" +
                "question='" + question + '\'' +
                ", answer='" + answer + '\'' +
                ", lastAnswerCorrect=" + lastAnswerCorrect +
                '}';
    }
}

File: Card.ts
export class Card {
    private question: string;
    private answer: string;
    private lastAnswerCorrect: boolean | null;

    constructor(question: string, answer: string) {
        this.question = question;
        this.answer = answer;
        this.lastAnswerCorrect = null;
    }

    getQuestion(): string {
        return this.question;
    }

    getAnswer(): string {
        return this.answer;
    }

    checkAnswer(userAnswer: string | null): boolean {
        if (userAnswer === null) {
            this.lastAnswerCorrect = false;
            return false;
        }

        const normalizedUser = userAnswer.trim();
        const correct = this.answer.toLowerCase() === normalizedUser.toLowerCase();
        this.lastAnswerCorrect = correct;
        return correct;
    }

    wasIncorrectInLastRound(): boolean {
        return this.lastAnswerCorrect === false;
    }

    toString(): string {
        return `Card{question='${this.question}', answer='${this.answer}', lastAnswerCorrect=${this.lastAnswerCorrect}}`;
    }
}
